scripts/generate_gitlab.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+import requests
+import yaml
+
+try:
+    gitlab_token = os.environ["GITLAB_TOKEN"]
+except KeyError:
+    print("Please provide the environment variable GITLAB_TOKEN")
+    sys.exit(1)
+
+parser = argparse.ArgumentParser(
+    description="Script to generate vcsconfig for all repositories \
+    under the given namespace (needs Gitlab >= 10.3)"
+)
+parser.add_argument("gitlab_host", type=str, help="url to the gitlab instance")
+parser.add_argument(
+    "gitlab_namespace",
+    type=str,
+    help="namespace/group in gitlab to generate vcsconfig for",
+)
+parser.add_argument(
+    "-c",
+    type=str,
+    help="path to the target config file (default: ./vcspull.yaml)",
+    dest="config_file_name",
+    required=False,
+    default="./vcspull.yaml",
+)
+
+args = vars(parser.parse_args())
+gitlab_host = args["gitlab_host"]
+gitlab_namespace = args["gitlab_namespace"]
+config_filename = args["config_file_name"]
```

Comment:
You might just want to create a [pathlib.Path](https://docs.python.org/3.10/library/pathlib.html) here, as it has easy object based handling (e.g. [Path.is_file()](https://docs.python.org/3.10/library/pathlib.html#pathlib.Path.is_file))

Reply:
@dvzrv @aschleifer  would either of you like to make a follow up PR improving upon this?

Reply:
@tony I will do a follow up PR soon to address this point.